packages/compiler-cli/src/ngtsc/typecheck/src/template_symbol_builder.ts

```diff
@@ -314,7 +313,8 @@ export class SymbolBuilder {
     // In either case, `_t1["index"]` or `_t1.index`, `node.expression` is _t1.
     // The retrieved symbol for _t1 will be the variable declaration.
     const tsSymbol = this.getTypeChecker().getSymbolAtLocation(node.expression);
-    if (tsSymbol === undefined || tsSymbol.declarations.length === 0 || selector === null) {
+    if (tsSymbol === undefined || tsSymbol.declarations === undefined ||
+        tsSymbol.declarations.length === 0 || selector === null) {
```

Comment:
```suggestion
    if (tsSymbol?.declarations === undefined ||
        tsSymbol.declarations.length === 0 || selector === null) {
```

I was tempted to suggest:

```suggestion
    if (!tsSymbol?.declarations.length > 0 || selector === null) {
```

But I think that is too obfuscated.

Reply:
Perhaps this would be acceptable?

```suggestion
    if (!tsSymbol?.declarations?.length || selector === null) {
```

Reply:
To me the first suggestion is the most readable. I'll change it to that.